pytorch_lightning/trainer/trainer.py

```diff
@@ -835,6 +838,10 @@ def _run(self, model: 'pl.LightningModule') -> Optional[Union[_EVALUATE_OUTPUT,
         # restore callback states
         self.checkpoint_connector.restore_callbacks()
 
+        if self.ckpt_path:
+            rank_zero_info(f"Loading checkpoint from {self.ckpt_path}")
+            self.checkpoint_connector.restore_model_weights(self.ckpt_path)
+
```

Comment:
Since there is a chance we're loading a checkpoint into a new Trainer object, we have to ensure the training type plugin has been connected to the Trainer.